### ISSUE TITLE
Show a gui message when graphics can't be initialized

### ISSUE THREAD

**Issue by bjorn3** [bevyengine/bevy#2389]
## What problem does this solve or what need does it fill?

When for example vulkan is not supported by the gpu or drivers are not installed bevy will currently panic with a message obscure to most users. In addition it won't be possible to see this message at all if the game was not launched in a terminal.

## What solution would you like?

Bevy should try to use a gui library like gtk to display a clear message to the user about the problem and with troubleshooting steps. As gtk may not be available, it may be a good idea to use `dlopen` to load it at runtime if an error happened. The error messages should not just be limited to failing to initialize vulkan, but any kind of failure thaf prevents showing anything on the screen. On platforms other than linux, it should preferably use the native gui library. This doesn't need dlopen of this guo library is guaranteed to exist on the respective platform.

## What alternative(s) have you considered?

Require every game developer to have their own troubleshooting guide and to deal with a lot of possibly trivially fixable issues that the average user may not find trivial.

**Comment by cart** [comment]
Yeah this seems like something we'll need to fix. But if it is at all possible, id rather not pull in a bunch of "native ui dependencies" for each platform we need to support. I wonder if theres a way to draw to a winit window without a gpu? We could then just rasterize error message text using our existing text rasterization logic.

**Comment by bjorn3** [comment]
> But if it is at all possible, id rather not pull in a bunch of "native ui dependencies" for each platform we need to support.

That was why I suggested dlopen. That way they will only be pulled in if graphics initialization failed.

> I wonder if theres a way to draw to a winit window without a gpu?

In a cross platform way? I don't think so. Winit only gives you a raw window handle. It is probably possible to do it for every platform separately. For some platforms I think winit has to pull in the native ui dependencies behind the scenes anyway, so you may as well directly use them. For example on macOS it uses core-graphics, which allows you to directly draw text. On wasm you can replace the `<canvas>` with a `<div>`.

**Comment by cart** [comment]
Makes sense. If theres no existing cross platform way to write a cpu defined image to a winit window and defining that logic would require pulling in the same "heft" of extra static dependencies that native ui messages would, then we might as well use the native UI libs.

**Comment by ksf** [comment]
Another idea would be to use `xmessage`, or a suitable platform equivalent. No need to `dlopen` or even choose one: Simply provide a mechanism with which developers can say "on critical error, `exec` into this command".
